lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp

```diff
@@ -7798,16 +7798,15 @@ void TypeSystemClang::AddMethodOverridesForCXXRecordType(
 
 std::unique_ptr<clang::CXXBaseSpecifier>
 TypeSystemClang::CreateBaseClassSpecifier(lldb::opaque_compiler_type_t type,
-                                          AccessType access, bool is_virtual,
-                                          bool base_of_class) {
+                                          AccessType access, bool is_virtual) {
   if (!type)
     return nullptr;
 
   return std::make_unique<clang::CXXBaseSpecifier>(
-      clang::SourceRange(), is_virtual, base_of_class,
+      clang::SourceRange(), is_virtual,
       TypeSystemClang::ConvertAccessTypeToAccessSpecifier(access),
       getASTContext().getTrivialTypeSourceInfo(GetQualType(type)),
-      clang::SourceLocation());
+      GetAsCXXRecordDecl(type), clang::SourceLocation());
```

Comment:
Ah - I think this might actually not be quite right. The second-to-last argument (i.e., `GetAsCXXRecordDecl(type)`) needs to be the `CXXRecordDecl` of the _derived_ class - as far as I can tell, `type` refers to the _base_ class.

Some context: In order to implement the [`is_accessible` metafunction](https://isocpp.org/files/papers/D2996R3.html#pnum_71) for base classes, it became necessary to have an "edge" from the `CXXBaseSpecifier` (which we internally use to represent a "reflection of a base class") back to the derived class. It's otherwise not possible to reason about e.g., whether a private base class is visible from a friend of the derived class.